### PR TITLE
Implement Retry logic in Salesforce batch source and Mutli Source Plugin for connection timeout issues

### PR DESCRIPTION
**Pull request by sgarg-CS**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java [hunk 1]
```diff
@@ -186,6 +207,21 @@ public String getOrgId(OAuthInfo oAuthInfo) throws ConnectionException {
     return partnerConnection.getUserInfo().getOrganizationId();
   }
 
+  @Nullable
+  public Long getInitialRetryDuration() {
+    return initialRetryDuration == null ? 5L : initialRetryDuration;
```
Comment (vikasrathee-cs) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java]: create variables for these in constants file for each property's default value

Comment (sgarg-CS) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java]: Added variables for all the constants used. ff723f92b5a5e5f66701811d4b1243dd38ddc732

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java [hunk 2]
```diff
@@ -186,6 +207,22 @@ public String getOrgId(OAuthInfo oAuthInfo) throws ConnectionException {
...
+  @Nullable
+  public Long getMaxRetryDuration() {
+    return maxRetryDuration == null ? SalesforceSourceConstants.DEFULT_MAX_RETRY_DURATION_SECONDS : maxRetryDuration;
+  }
+
+  @Nullable
+  public Integer getMaxRetryCount() {
+    return maxRetryCount == null ? SalesforceSourceConstants.DEFAULT_MAX_RETRY_COUNT : maxRetryCount;
+  }
+
```
Comment (psainics) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java]: @ Nullable is not required here, also we can use primitive to better show that.

Comment (psainics) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java]: @Shubhangi-cs Please update these.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java [hunk 3]
```diff
@@ -188,6 +219,19 @@ public String getOrgId(OAuthInfo oAuthInfo) throws ConnectionException {
...
+  public Long getMaxRetryDuration() {
+    return maxRetryDuration == null ? SalesforceSourceConstants.DEFULT_MAX_RETRY_DURATION_SECONDS : maxRetryDuration;
+  }
+
+  public Integer getMaxRetryCount() {
+    return maxRetryCount == null ? SalesforceSourceConstants.DEFAULT_MAX_RETRY_COUNT : maxRetryCount;
+  }
+
```
Comment (psainics) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java:234]: @Shubhangi-cs use can primitive types here.

Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java:234]: This is fine

**Pull request merged**